### ISSUE TITLE
KafkaStreamsITCase is flaky causing endless TimeoutException warnings leading to job being aborted

### ISSUE THREAD

**Issue by famod** [quarkusio/quarkus#16415]
E.g. [here](https://github.com/quarkusio/quarkus/runs/2309475289?check_suite_focus=true) and multiple times in #16249:
```
2021-04-09 21:29:04,547 INFO  [io.quarkus] (main) Profile prod activated. 
2021-04-09 21:29:04,547 INFO  [io.quarkus] (main) Installed features: [cdi, kafka-streams, micrometer, resteasy, resteasy-jackson, smallrye-health]
2021-04-09 21:29:14,537 ERROR [io.qua.kaf.str.run.KafkaStreamsProducer] (pool-4-thread-1) Failed to get topic names from broker: java.util.concurrent.TimeoutException
	at org.apache.kafka.common.internals.KafkaFutureImpl$SingleWaiter.await(KafkaFutureImpl.java:108)
	at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:272)
	at io.quarkus.kafka.streams.runtime.KafkaStreamsProducer.waitForTopicsToBeCreated(KafkaStreamsProducer.java:306)
	at io.quarkus.kafka.streams.runtime.KafkaStreamsProducer.access$000(KafkaStreamsProducer.java:60)
	at io.quarkus.kafka.streams.runtime.KafkaStreamsProducer$1.run(KafkaStreamsProducer.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
```
The same can also happen in `KafkaSnappyProducerITCase`, e.g. [here](https://github.com/quarkusio/quarkus/runs/2309287178?check_suite_focus=true).

There seems to be a configuration issue, because before those timeouts the log is flooded with warnings like these (just en excerpt):
```
[INFO] Running io.quarkus.it.kafka.streams.KafkaStreamsITCase
2021-04-09 22:08:31,378 WARN  [io.qua.deployment] (main) Producing values from constructors and fields is no longer supported and will be removed in a future release: io.quarkus.deployment.annotations.BuildProducer io.quarkus.it.classtransformer.ClassTransformerProcessor.transformers
Executing [/home/runner/work/quarkus/quarkus/integration-tests/kafka-streams/target/quarkus-integration-test-kafka-streams-999-SNAPSHOT-runner, -Dquarkus.http.port=8081, -Dquarkus.http.ssl-port=8444, -Dtest.url=http://localhost:8081, -Dquarkus.log.file.path=/home/runner/work/quarkus/quarkus/integration-tests/kafka-streams/target/quarkus.log, -Dquarkus.log.file.enable=true]
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-04-09 22:08:38,753 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.truststore.location' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'sasl.login.refresh.buffer.seconds' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.truststore.password' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.truststore.type' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.endpoint.identification.algorithm' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'sasl.kerberos.ticket.renew.jitter' was supplied but isn't a known config.
2021-04-09 22:08:38,789 WARN  [org.apa.kaf.str.pro.int.StateDirectory] (main) Using /tmp directory in the state.dir property can cause failures with writing the checkpoint file due to the fact that this directory can be cleared by the OS
```

**Comment by cescoffier** [comment]
Hopefully fixed by #16435

**Comment by famod** [comment]
Thanks @cescoffier! I'll close this one. In case the problem pops up again we can always reopen this issue.
